src/main/kotlin/bot/impl/DiscordBotConnection.kt

```diff
@@ -49,8 +50,17 @@ class DiscordBotConnection(private val config: Config) : BotConnection, Listener
         this.jda = JDABuilder.createDefault(this.config[DiscordBotSpec.token])
                 .addEventListeners(this)
                 .build()
-        if (this.config[DiscordBotSpec.webhook].isNotEmpty()) {
-            this.webhookClient = WebhookClientBuilder(this.config[DiscordBotSpec.webhook])
+        if (this.config[DiscordBotSpec.chatWebhook].isNotEmpty()) {
+            this.chatWebhookClient = WebhookClientBuilder(this.config[DiscordBotSpec.chatWebhook])
+                .setDaemon(true)
+                .setAllowedMentions(allowedMentions)
+                .setHttpClient(OkHttpClient.Builder()
+                    .protocols(listOf(Protocol.HTTP_1_1))
+                    .build())
+                .build()
+        }
+        if (this.config[DiscordBotSpec.eventWebhook].isNotEmpty()) {
+            this.eventWebhookClient = WebhookClientBuilder(this.config[DiscordBotSpec.eventWebhook])
                 .setDaemon(true)
```

Comment:
The new webhook client should also use the
```kotlin
.setHttpClient(OkHttpClient.Builder()
                    .protocols(listOf(Protocol.HTTP_1_1))
                    .build())
```
to prevent a ~5 min hang on server stop (see #8)
(using the same instance should work for both)

Reply:
I... could have sworn I fixed that when I merged the upstream changes, oops. I'll make the changes when I wake up

Reply:
actually, it seems it IS using the .setHttpClient code, it just doesn't *look like it is* because of how the git diff appears - the .setHttpClient can be seen just at the end of the diff in github's diff viewer, and the full chain is visible in the file proper.

Reply:
ah yeah it is, my bad